module/platform/darwin/HardwareIO/src/darwin/Darwin.cpp

```diff
@@ -129,10 +130,11 @@ namespace Darwin {
 
         // Do a self test so that we can move all the sensors that are currently failing to the end of the list
         std::vector<std::pair<uint8_t, bool>> sensors = selfTest();
-        std::sort(
-            std::begin(sensors),
-            std::end(sensors),
-            [](const std::pair<uint8_t, bool>& a, const std::pair<uint8_t, bool>& b) { return a.second > b.second; });
+        std::sort(std::begin(sensors),
+                  std::end(sensors),
+                  [](const std::pair<uint8_t, bool>& a, const std::pair<uint8_t, bool>& b) {
+                      return static_cast<int>(a.second) > static_cast<int>(b.second);
```

Comment:
What is the reason on not using the `unit8_t` comparison

Reply:
looking at this closer, it seems to be doing a `greater than` on the bool, which is implicitly cast to integer, which clang-tidy doesn't like. This seems like it should be a `std::partition` instead.